corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/ontonotes/OntonotesTreebankReader.java

```diff
@@ -0,0 +1,225 @@
+            throw new IllegalStateException(e);
+        }
+    }
+
+    /**
+     * parse the pen treebank parse file, producing an annotation covering the entire file.
+     * @param lines the data from the file, each line.
+     * @return the text annotation.
+     * @throws AnnotatorException
+     */
+    private TextAnnotation findNextTree(ArrayList<String> lines) throws AnnotatorException {
+        StringBuilder sb = new StringBuilder();
+        int numParen = 0;
+        int currentLineId = 0;
+        ArrayList<String[]> sentences = new ArrayList<> ();
+        ArrayList<Tree<String>> trees = new ArrayList<> ();
+        while (currentLineId < lines.size()) {
+            String line = lines.get(currentLineId++);
+            if (line.length() == 0)
+                continue;
+            
+            numParen += countUnclosedParens(line);
+            sb.append(line);
+            if (numParen == 0) {
+                // parse the tree, add the sentence tokens to the list of sentences.
+                Tree<String> tree = TreeParserFactory.getStringTreeParser().parse(
+                    sb.toString().replaceAll("\\\\/", "/"));
+                
+                // get the tokens.
+                String[] text = ParseUtils.getTerminalStringSentence(tree);
+                if (text.length != 0) {
+                    sentences.add(text);
+                    trees.add(tree);
+                } else 
+                    System.err.println("Y "+tree);
+                sb = new StringBuilder();
+            }
+        }
+
+        TextAnnotation ta = BasicTextAnnotationBuilder.createTextAnnotationFromTokens(
+            PENN_TREEBANK_ONTONOTES, currentfile, sentences);
+        TreeView parse = new TreeView(parseViewName, "PTB-GOLD", ta, 1.0);
+        
+        // add each parse tree
+        int treecount = 0;
+        for (Tree<String> tree : trees)
+            parse.setParseTree(treecount++, tree);
+        ta.addView(parseViewName, parse);
+        POSFromParse pos = new POSFromParse(parseViewName);
+        ta.addView(pos);
+        return ta;
+    }
+
+    /**
+     * count the number of unclosed opening parens, this can be negative if there are more closing
+     * parens that opening ones.
+     * @param line the line of data.
+     * @return the number of unclosed parens.
+     */
+    private int countUnclosedParens(String line) {
+        int unclosedParens = 0;
+        for (char character : line.toCharArray()) {
+            if (character == ')') unclosedParens--;
+            else if (character == '(') unclosedParens++;
+        }
+        return unclosedParens;
+    }
+
+    /**
+     * TODO: generate a human-readable report of annotations read from the source file (plus whatever
+     * other relevant statistics the user should know about).
+     */
+    public String generateReport() {
+        throw new UnsupportedOperationException("ERROR: generateReport() Not yet implemented.");
```

Comment:
report number of trees read; also, any failures if they can occur (e.g. format exception or some such)